### PR TITLE
Implement cache and sparse-set as Janet abstract type

### PR DESCRIPTION
**Pull request by pepe**
I started with the implementation discussed in #25 and #21. I implemented only the cache in barebone, so we can confirm it works right (no leaks, etc.) and check the performance improvements (if any).

Sorry, I have not waited for the answers, but I found some time between work :-).

### CODE REVIEW

**Comment by pepe** [comment]
I have fixed the location of the native library when installed by jpm.

OT: performance test is failing for me on `unknown symbol print-results`

**Comment by pepe** [comment]
I added an abstract type `next` method to enumerate the cache better.

**Comment by pepe** [comment]
At the moment, I guess the `cache/clear` function is enough no need for the whole abstract methods machinery. 

What do you think? For sparse-set, it is another story indeed.

**Comment by pepe** [comment]
Also if you would not mind another developer only dependency, I have this for benchmarking https://git.sr.ht/~pepe/paula/tree/master/item/paula/bench.janet#L25 :-)

**Comment by AlecTroemel** [comment]
> Also if you would not mind another developer only dependency, I have this for benchmarking https://git.sr.ht/~pepe/paula/tree/master/item/paula/bench.janet#L25 :-)

I wonder if that something bakpakin would merge into spork. In any case, adding it as a dep here is fine with me!.. Though my gut says we'll have problems with the macros in `stress-test.janet`, given how un-hygenic they are :laughing:

File: project.janet [hunk 1]
```diff
@@ -7,5 +7,16 @@
   :repo "git+https://github.com/AlecTroemel/junk-drawer"
   :dependencies ["spork"])
 
+
+(add-loader)
+(import /src/cache)
+
+(task "cache.c" []
+      (cache/render "cache.c"))
```
Comment (AlecTroemel) [inline project.janet]: could we put the rendered C file in `src/`, instead of the root of the project?

**Comment by AlecTroemel** [comment]
> I have fixed the location of the native library when installed by jpm.
> 
> OT: performance test is failing for me on `unknown symbol print-results`

I _think_ thats a holdover from another package I was trying out, and just forgot to remove :facepalm:

File: project.janet (hunk 1 above)
Comment (pepe) [reply project.janet]: Indeed. It should be ignored anyway.

**Comment by pepe** [comment]
> Though my gut says we'll have problems with the macros in stress-test.janet, given how un-hygenic they are

I would not be so hard on ourselves 😜

**Comment by pepe** [comment]
I am finishing the summer season in the garden, so today and most of tomorrow AFK.

**Comment by pepe** [comment]
I have done some benchmarks of this approach on another project, and the improvements are minimal. I guess it is not worth the effort.

Would you mind if I looked at https://github.com/AlecTroemel/janet-chipmunk/issues/1?

**Comment by pepe** [comment]
Maybe it was premature optimization. Sorry for the press :-). My surname in Czech means something like "the man in a hurry" :-).

**Comment by AlecTroemel** [comment]
@pepe all good! I think it makes sense to leave `cache.janet` as is for now, given the lack of performance improvements. I'll let you close this PR if thats agreeable

**Pull request closed without merging**